core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -617,8 +667,7 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     extends org.apache.kafka.clients.consumer.ConsumerRebalanceListener {
 
     override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]) {
-      producer.flush()
-      commitOffsets(mirrorMakerConsumer)
+      mirrorMakerConsumer.requestAndWaitForCommit()
```

Comment:
We do not need the synchronization in new consumer. This will cause deadlock because there is no separate thread doing rebalance in the new consumer.